js/packages/nugget-logger/index.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('fs');
+
+const levels = [
+    'ERROR',
+    'WARN',
+    'INFO',
+    'VERBOSE',
+    'DEBUG',
+    'SILLY'
+];
+
+class logger {
+
+    /**
+     * @param level - name (not index) of the level for this logger to be
+     * @param logFilePath - file name/path for file to write logs to
+     */
+    constructor(level, logFilePath = '') {
+        // make sure the level is between 0 and 5
+        level = levels.indexOf(level.toUpperCase());
+        if (level < 0 || level > levels.length - 1)
+            throw 'level must be between 0 and 5 inclusice';
+
+        // this.level is set to a number instead of the actual name of the level
+        // the number is the index of the level name in `levels`
+        this.level = level;
+
+        this.logFile = logFilePath
+            ? fs.createWriteStream(logFilePath, { flags: 'a' })
+            : this.logFile = {
+                write: () => {}
+            };
+    }
+
+    print(label, message, level) {
+        // don't log anything above the level we specified in the constructor
```

Comment:
` // don't log anything above the level we specified in the constructor `

Is this to allow for different levels of verbosity, or something else? Why do we want this feature?

Reply:
Yeah it's so we can change the level of verbosity. We don't do anything with this yet but it can be specified in script arguments & stuff in the future.